clash-prelude/tests/Clash/Tests/Vector.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Clash.Tests.Vector where
+
+import           Clash.Sized.Vector (Vec((:>), Nil))
+import           Clash.XException
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+import           Test.Tasty.TH
+
+i :: Int -> Int
+i = id
+
+case_showXVector :: Assertion
+case_showXVector = "<1,2,X" @=? showX (1 :> i 2 :> errorX "def")
+
+case_showX2DVector :: Assertion
+case_showX2DVector = "<<1,X,<3,5" @=? showX ((1 :> errorX "def") :> (3 :> i 5 :> Nil) :> Nil)
```

Comment:
Why doesn't this print as:
```
<<1,X,<3,5>>
```

Reply:
I think https://github.com/clash-lang/clash-compiler/pull/1779/files#diff-0983c302bba1b7e92851789687ff742546e7453d6b5b9cf032076e6131fa0599R321, but I was just pulling locally to check

Reply:
Ah, right, that should probably become:
```haskell
punc (Cons x (isX -> Right Nil)) = showsX x . ('>':)
```

Reply:
I don't think the tests are even being run! When I change the test to:

```haskell
case_showX2DVector = "penguins" @=? showX ((1 :> errorX "def") :> (3 :> i 5 :> Nil) :> Nil) 
```

it still passes. Which seems _slightly_ wrong

Reply:
Good point Alex, it hasn't been added to the file collecting all tests.

Reply:
It has now :P